paddle/fluid/operators/controlflow/conditional_block_op.cc

```diff
@@ -204,7 +325,8 @@ class ConditionalBlockGradOp : public ConditionalOp {
     for (size_t i = 0; i < outside_grads.size(); ++i) {
       const std::string &outside_grad_name = outside_grads[i];
       const std::string &inside_grad_name = inside_grads[i];
-      VLOG(4) << "inside_grad_name = " << inside_grad_name
+      VLOG(4) << "[assign local]"
```

Comment:
For `ConditionalBlockOp`, only if the condition meets, its block will be executed and we can get the corresponding gradient. In principle, we do not need to do anything special. However, when appending grad block, unlike forward block, paddle may add duplicate variable. So only the variables in the local scope of grad block will be set and we need to copy their values to outer scope. This is what `AssignLocalGradientToParentScope` does.

I changed the log here to make it more clear because it's similar with the log in `AssignZeroToParentScope`.